pkg/bazel/bazel_flags.go

```diff
@@ -138,13 +167,178 @@ func (b *bazel) AddBazelFlags(cmd *cobra.Command) error {
+
+		// Complete packages relative to pwd.
+		default:
+			bzlPackages, err := b.expandPackageNames(bzlPackage, true)
+			if err != nil {
+				return nil, cobra.ShellCompDirectiveError
+			}
+			return bzlPackages, cobra.ShellCompDirectiveNoSpace
+		}
+	}
+}
+
+func (b *bazel) expandPackageNames(bzlPackage string, recurse bool) ([]string, error) {
+	trailingSlash := strings.HasSuffix(bzlPackage, "/")
+
+	// Do not recurse if we are completing a package with trailing slash. The
+	// user has indicated they expect sub-packages in the provided package.
+	recurse = recurse && !trailingSlash
+
+	entries, err := os.ReadDir(bzlPackage)
+	if err != nil && (!errors.Is(err, os.ErrNotExist) || !recurse) {
+		return nil, err
+	}
+	// Directory does not exist, complete with help of the parent.
+	if err != nil {
+		return b.expandPackageNames(filepath.Dir(bzlPackage), false)
+	}
+
+	var (
+		hasBuildFile bool
+		bzlPackages  []string
+	)
+	for _, e := range entries {
+		name := e.Name()
+		// If build file exists, we will suggest "<package>:" for convenience.
+		if name == "BUILD" || name == "BUILD.bazel" {
+			hasBuildFile = true
+		}
+		// Only directories can be packages.
+		if !e.IsDir() {
+			continue
+		}
+		// Skip symlinks (e.g. bazel-bin)
+		if e.Type()&os.ModeSymlink != 0 {
+			continue
+		}
+		// Skip dotted directories (e.g. .git)
+		if strings.HasPrefix(name, ".") {
+			continue
+		}
+		bzlPackages = append(bzlPackages, filepath.Join(bzlPackage, name))
+	}
+	// Only create the the convenience "<package>:" if the user does not
+	// indicate that they want a sub-package.
+	if hasBuildFile && !trailingSlash {
+		bzlPackages = append(bzlPackages, func() string {
+			if bzlPackage == "." {
+				return ":"
+			}
+			return bzlPackage + ":"
+		}())
+	}
+	return bzlPackages, nil
+}
+
+func listBazelRules(ctx context.Context, bzlPackage string) ([]string, error) {
+	var stdout bytes.Buffer
+	var stderr strings.Builder
+	opts := &edit.Options{
+		OutWriter: &stdout,
+		ErrWriter: &stderr,
+		NumIO:     200,
+	}
+	if ret := edit.Buildozer(opts, []string{"print label", bzlPackage + ":all"}); ret != 0 {
+		return nil, fmt.Errorf("buildozer exit %d: %s", ret, stderr.String())
+	}
+
+	rules := strings.Split(strings.TrimSpace(stdout.String()), "\n")
+
+	// Do post-processing on the rules. If the label is equal to the package,
+	// it is reported in the short form without colon. Make sure to use the same
+	// path as provided in bzlPackage, even if buildozer resolves to a fully
+	// qualified label in the workspace.
+	for i, t := range rules {
+		if _, label, ok := strings.Cut(t, ":"); ok {
+			rules[i] = bzlPackage + ":" + label
+			continue
+		}
+		rules[i] = bzlPackage + ":" + path.Base(t)
+	}
+
+	return rules, nil
+}
+
+func listBazelFlags(command string) []string {
+	bazelFlags, ok := bazelFlagSets[command]
+	if !ok {
+		return nil
+	}
+	var flags []string
+	bazelFlags.VisitAll(func(f *pflag.Flag) {
+		flags = append(flags, "--"+f.Name)
```

Comment:
hmm. suppose the `--noable` flags should be handled as well...

Reply:
The --noable flags should be take care of:

```
bazel run //cmd/aspect --  __complete build "--"  | grep '\-\-no' 
...
--notrim_test_configuration
--nouse_action_cache
--nouse_ijars
--nouse_singlejar_apkbuilder
--nouse_top_level_targets_for_symlinks
--nouse_workers_with_dexbuilder
--noverbose_explanations
--noverbose_failures
--nowatchfs
--noworker_quit_after_build
--noworker_sandboxing
--noworker_verbose
--nozip_undeclared_test_outputs
```